Assets/Scripts/Actors/BuddyStats.cs

```diff
@@ -13,27 +27,80 @@ public bool isAlive
 		get { return _isAlive; }
 	}
 
-	int currentStats = 0;
 	ParticleSystem _particles;
 
+	uint ID = 0;
+
 	void Awake()
 	{
+		ID = GetID(); // Grab the current unique ID
+		name = "Buddy " + GetRandomName( ID );
+		currResourceTimer = decreaseResourcesTime;
+		apples = startingApples;
 		_particles = GetComponentInChildren<ParticleSystem>();
+		SetGod(owner); // For owners set in the inspector
+	}
+
+	public void SetGod( GodTag _godTag )
+	{
+		if(_godTag)
+		{
+			owner = _godTag;
+			_actorStats = _godTag.gameObject.GetComponent<PlayerActorStats>();
+		}
+	}
+
+	static string[] names = { "Longnose", "Jojo", "JillyJane", "Sunshine", "Moosejaw"};
```

Comment:
I'm not sure what the best way to handle this is. BuddyStats isn't a playerActorComponent on the God so it wouldn't work to use the reference there. I do think this should all be wrapped in a property though.